botorch/acquisition/knowledge_gradient.py

```diff
@@ -87,11 +87,11 @@ def __init__(
             sampler: The sampler used to sample fantasy observations. Optional
                 if `num_fantasies` is specified.
             objective: The objective under which the samples are evaluated. If
-                `None` or a ScalarizedObjective, then the analytic posterior mean
-                is used, otherwise the objective is MC-evaluated (using
-                inner_sampler).
+                `None`, then the analytic posterior mean is used. Otherwise, the
+                objective is MC-evaluated (using inner_sampler).
+            posterior_transform: An optional PosteriorTransform.
```

Comment:
can you explain how this is used with the objective (i.e. transform, then sample, then application of objective)?